platform-sdk/docs/core/tipset-algorithm.md

```diff
@@ -0,0 +1,260 @@
+across all tipsets being merged, and constructing the resulting tipset using those generations.
+
+```
+Example
+
+[1, 3, 5, 2] merged with [7, 3, 2, 11] == [7, 3, 5, 11]
+```
+
+To compute the tipset of a new event, merge the tipsets of the event's parents,
+and then update the generation for the node's creator.
+
+```
+Example
+
+In a 4 node network with nodes A, B, C, and D, suppose B is creating a new event. The self parent has a generation
+3 and a tipset [1, 3, 5, 2]. The other parent is created by node D, has a generation of 11,
+and a tipset of [7, 3, 5, 11].
+
+The new node has a generation of 12 (computed via max(3, 11) + 1). The merged tipsets of the parents results in a tipset
+of [7, 3, 5, 11]. Then, update the self generation to 12 and get [7, 12, 5, 11].
+```
+
+# Tipset Advancement Score
+
+A tipset advancement score between two tipsets X and Y is defined as the number of entries in tipset Y that are
+strictly greater than the corresponding entry in tipset X.
+
+
+```
+Example
+
+X = [1, 3, 5, 2]
+Y = [7, 3, 2, 11]
+
+For computing the advancement score between X and Y, look at each pair of generations:
+
+1 vs 7: 7 > 1, so this is an advancement
+3 vs 3: 3 == 3, so this is not an advancement
+5 vs 2: 2 < 5, so this is not an advancement
+2 vs 11: 11 > 2, so this is an advancement
+
+The total advancement count from X to Y is therefore 2. Note that the amount an entry advances is not important,
+only if it increased or did not increase.
+```
+
+# Weighted Tipset Advancement Score
+
+A weighted tipset advancement score is very similar to a tipset advancement score. But instead of adding 1 for
+each entry that has a greater value, instead add the amount of consensus weight held by that node.
+
+```
+Example
+
+Suppose an address book has the following weights: A = 5, B = 9, C = 11, D = 2
+
+Find the weighed advancement score between X and Y.
+
+X = [1, 3, 5, 2]
+Y = [7, 3, 2, 11]
+
+The entry for node A advances, add 5. The entry for D advances, add 2. Total weighed advancement score is 7.
+```
+
+# Partial Weighted Tipset Advancement Score
+
+This is similar to the weighed advancement score with one minor tweak. A partial weighted advancement score
+is always calculated from the perspective of a particular node. When computing a partial advancement score,
+ignore the advancement provided by a target node.
+
+```
+Example
+
+Suppose an address book has the following weights: A = 5, B = 9, C = 11, D = 2
+
+Find the partial weighted advancement score between X and Y with a target of node A.
+
+X = [1, 3, 5, 2]
+Y = [7, 3, 2, 11]
+
+The entry for node A advances, but we ignore it because A is the target.
+The entry for D advances, add 2. Total weighed advancement score is 2.
+```
+
+# Snapshot Improvement Scores
+
+Each new event is assigned a tipset improvement score. Each node assigns the events that it creates an
+improvement score. Nodes do not assign events created by other nodes an improvement score.
```

Comment:
By shortening `snapshot improvement score` to `improvement score`, the reader must be able to differentiate between `improvement score`, and `advancement score` mentioned above.

I think these terms might be too similar. I would say either always use the unshortened term,or perhaps define a less similar term

Reply:
Renamed this section to `Advancement Score Improvement Relative to the Snapshot`. Also reworded the text. Instead of attempting to define a new term, I try and make it clear that I am using the previous terms, just relative to the special tipset called the snapshot.